File: 0206-reverse-linked-list/0206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) 
{
    struct ListNode* prevNode = 0;
    struct ListNode* curNode = head;
    struct ListNode* nextNode = head;

    while(curNode != 0)
    {
        nextNode = nextNode->next;
        curNode->next = prevNode;
        prevNode = curNode;
        curNode = nextNode;
    }

    head = prevNode;
    return head;
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curNode = head
        nextNode =  head
        prevNode = None
        while(curNode != None):
            nextNode = nextNode.next
            curNode.next = prevNode
            prevNode = curNode
            curNode = nextNode
        head = prevNode
        return head